common/dpcpp/slab_hash.hpp

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include <CL/sycl.hpp>
+#include <algorithm>
+#include "dpcpp_common.hpp"
+
+#define SUBGROUP_SIZE 16
+#define CONST 64
+#define SLAB_SIZE CONST * SUBGROUP_SIZE
+
+#define CLUSTER_SIZE 1024
+
+#define BUCKETS_COUNT 128
+
+#define EMPTY_UINT32_T 4294967295
+
+using std::pair;
+
+template <size_t A, size_t B, size_t P>
+struct Hasher {
```

Comment:
Hasher might be too general of a name, please consider wrapping into a namespace